lib/protocol/events.js

```diff
@@ -273,28 +282,68 @@ function eventsProtocolFactory (
         );
     };
 
-    EventsProtocol.prototype.publishExternalEvent = function (data) {
-        assert.object(data);
-        assert.string(data.type);
-        assert.string(data.action);
-        assert.string(data.severity);
-        assert.string(data.typeId);
-        assert.ok(data.hasOwnProperty("payload"));
-        if (data.nodeId !== null){
-            assert.string(data.nodeId);}
-
-        data.version = '1.0';
-        data.createdAt = new Date();
-
-        var routingKey = data.type + '.' + data.typeId + '.' + data.action +
-                         '.' + data.severity + '.' + data.nodeId;
+    EventsProtocol.prototype.publishExternalEvent = function (eventData) {
+        assert.object(eventData);
+        assert.string(eventData.type);
+        assert.string(eventData.action);
+        assert.string(eventData.severity);
+        assert.string(eventData.typeId);
+        assert.ok(eventData.hasOwnProperty("data"));
+
```

Comment:
suggest to add hint for err msg here, like 
```
assert.string(eventData.typeId, 'typeId');
```
In this case, if the parameter doesn't contain typeId, the error msg would be "typeId (string) is required".
In current situation, the error msg is "undefined (string) is required", which is difficult for debugging.